File: backend/schemas/strategy.py
from pydantic import BaseModel


class RecommendationItem(BaseModel):
    ticker: str
    prediction: str
    confidence: int
    reason: str


class RecommendationsResponse(BaseModel):
    date: str
    recommendations: list[RecommendationItem]


class BacktestResponse(BaseModel):
    strategy_id: str
    returns: list[float]
    benchmark: list[float]
    dates: list[str]

File: frontend/src/types/strategy.ts
export interface Recommendation {
  ticker: string
  prediction: 'up' | 'down'
  confidence: number
  reason: string
}

export interface RecommendationsResponse {
  date: string
  recommendations: Recommendation[]
}

export interface BacktestResponse {
  strategy_id: string
  returns: number[]
  benchmark: number[]
  dates: string[]
}
